pkg/cmd/cli/cmd/tag.go

```diff
@@ -389,6 +389,9 @@ func (o TagOptions) RunTag() error {
 			default:
 				istag.Tag.From.Name = localRef.NameString()
 				istag.Tag.From.Namespace = o.ref.Namespace
+				if len(o.ref.Namespace) == 0 {
```

Comment:
So earlier code should be checking whether the target the user provided matches the current namespace - if it does, it is omitted when you set the ObjectReference (no ObjectReference should have Namespace != 0 if Namespace == object Namespace that contains the ref)